### DISCUSSION TITLE
RKE2 on Linux restarts with previous state

### DISCUSSION THREAD

**Discussion by threeseed** [rancher/rke2#4405] [Q&A]
**Environmental Info:**
RKE2 Version: 
v1.25.11+rke2r1

Node(s) CPU architecture, OS, and Version: 
Linux harana-server 5.15.0-75-generic #82-Ubuntu SMP Tue Jun 6 23:10:23 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration:
1 server, 2 agents (Linux and Windows)

**Describe the bug:**
When restarting RKE2 it loads with the previous state e.g. the uptime on Pods, Containers, Nodes are not reset back to zero.

**Steps To Reproduce:**
Installed RKE2 using [Quick Start](https://docs.rke2.io/install/quickstart).

Previous State:
<img width="1800" alt="Screenshot 2023-07-04 at 7 49 00 am" src="https://github.com/rancher/rke2/assets/657041/4fc07524-ee88-415d-a70f-bdabceca301f">

Stopped RKE2:
```
systemctl stop rke2-server.service
```

But RKE2 is not stopped:
<img width="1800" alt="Screenshot 2023-07-04 at 7 49 16 am" src="https://github.com/rancher/rke2/assets/657041/f5a10d88-ab07-4f8e-ac30-b6295846fcec">

Manually kill all RKE2 processes:
```
pkill -f rke2
```

RKE2 is now stopped so we restart it with:
```
systemctl start rke2-server.service
```

<img width="1800" alt="Screenshot 2023-07-04 at 7 50 16 am" src="https://github.com/rancher/rke2/assets/657041/93954d39-7c98-4b4a-bd74-f893d1f17b69">

New State:
<img width="1800" alt="Screenshot 2023-07-04 at 7 50 31 am" src="https://github.com/rancher/rke2/assets/657041/4bb36307-c52b-4072-bf36-564c4ff85bcc">

**Expected behavior:**
Uptime should be zero. IP address should change etc.

**Comment by brandond** [comment]
You appear to be a little confused - stopping the RKE2 service does not stop the pods. They continue running in the background, and should be unaffected as long as they can tolerate an outage of the local apiserver. This is why the pod uptime and IPs do not reset. If you want to kill the pods, use the rke2-killall.sh script.

**Comment by threeseed** [comment]
So I have run the systemctl stop and rke2-killall scripts and it is still showing the nodes as being up for 2d as opposed to zero.

Is there another script to stop the server node itself ?

**Reply by brandond** [reply]
I suspect you're looking at the age field on the node? That is how long it has been in the cluster. It is not uptime.

**Reply by threeseed** [reply]
Bit odd but sure.

So then I have modified /etc/rancher/rke2/config.yaml .. how do I update an existing cluster to reload it.

**Reply by brandond** [reply]
> Bit odd but sure.

This is not RKE2 specific, this is just how Kubernetes works.

> how do I update an existing cluster to reload it.

Just restart the rke2 service on the node where you changed the config.
